language-server/src/server.js

```diff
@@ -135,6 +144,37 @@ connection.listen();
 
 const documents = new TextDocuments(TextDocument);
 
+// CONFIGURATION
+
+const documentSettings = new Map();
+let globalSettings = {};
+
+async function getDocumentSettings(resource) {
+  if (!hasConfigurationCapability) {
+    return globalSettings;
+  }
+
+  if (!documentSettings.has(resource)) {
+    const result = await connection.workspace.getConfiguration({
+      scopeUri: resource,
+      section: "jsonSchemaLanguageServer"
+    });
+    documentSettings.set(resource, result ?? globalSettings);
+  }
+
+  return documentSettings.get(resource);
+}
+
+connection.onDidChangeConfiguration((change) => {
+  if (hasConfigurationCapability) {
+    documentSettings.clear();
+  } else {
+    globalSettings = change.settings.jsonSchemaLanguageServer || globalSettings;
```

Comment:
This should probably use `??`, right?

Reply:
Ah, yes. I think we had a confusion about when to use global settings back then.